src/event_parser.rs

```diff
@@ -381,404 +357,222 @@ mod tests {
         evt
     }
 
-    use futures::{stream, task::noop_waker, Stream};
-
-    fn one_chunk(bytes: &[u8]) -> impl Stream<Item = Result<Bytes>> + Unpin {
-        let chunk = chunk(bytes);
-        stream::iter(iter::once(Ok(chunk)))
-    }
-
-    fn delay_one_poll<T, E>() -> impl Stream<Item = core::result::Result<T, E>> + Unpin {
-        let mut ready = false;
-        stream::poll_fn(move |_cx| {
-            if !ready {
-                ready = true;
-                return Poll::Pending;
-            }
-            Poll::Ready(None)
-        })
-        .fuse()
-    }
-
-    fn delay_one_then<T, E>(t: T) -> impl Stream<Item = core::result::Result<T, E>> + Unpin {
-        delay_one_poll().chain(stream::iter(iter::once(Ok(t))))
-    }
-
     #[test]
     fn test_event_without_data_yields_no_event() {
-        use Poll::Ready;
-        let waker = noop_waker();
-        let mut cx = Context::from_waker(&waker);
-
-        let mut decoded = Decoded::new(one_chunk(b"id: abc\n\n"));
-        assert_eq!(decoded.poll_next_unpin(&mut cx), Ready(None));
+        let mut parser = EventParser::new();
```

Comment:
Not having the event parser be a stream really makes these tests much more straight forward.